### ISSUE TITLE
An error!

### ISSUE THREAD

**Issue by burhanbarakli** [PKU-ICST-MIPL/DRLIH_TMM2020#4]
in below code in test_util.py, list.sort(cmp) gives an error for list arguments! Is it necessary sorting ? Can i change it to comment line.

And the other question is when does the Train end? Because in train.py while condition is true, it goes infinity! So how many episode run for article results? 

When should the while loop terminate? Because every 10000 episodes low_lr function works.

In short, according to your experience, how many episodes do I have to try?

Thanks for response!

```
def cmp(a,b):
	if a[0]==b[0]:
		return 0
	if a[0]>b[0]:
		return 1
	if a[0]<b[0]:
		return -1

def get_map(query,data,query_lab,data_lab):
	tot = 0
	print 
	for i in range(len(query)):
		list = []
		eletot = 0.0
		cnt = 0
		for j in range(len(data)):
			list .append(  (calc(query[i],data[j]),data_lab[j])  )
		list.sort(cmp)  ##########################################This code gives error

		for j in range(len(list)):
			if list[j][1]==query_lab[i]:
				cnt += 1
				eletot += cnt*1.0/(1.0+j)
		eletot /= cnt
#		print list
#		print eletot
		tot += eletot
	tot /= len(query)
	return tot
```

**Comment by burhanbarakli** [comment]
`	Dtest=Dataloader("/home/yezhaoda/TMM-DRLIH/cifar10/resize64-rf-noncrop/test",0,5500,0,10)
	traintest=Dataloader("/home/yezhaoda/TMM-DRLIH/cifar10/resize64-rf-noncrop/train",0,500,0,10)`

the other question,5000 images in train and the other 55000 images for test! An i right?

**Comment by magicfisk** [comment]
this code is runing with python 2.7, the code might gives errors with pyrhon 3. 

in our experiment, the best performance is 30,000-40,000 episodes on cifar dateset.

**Comment by chj123456789** [comment]
请问函数cmp的作用是？我在运行时将它删除了，导致我测试时map总是很低

**Comment by magicfisk** [comment]
> 

您好。cmp是比较函数，用于对检索结果的排序。list中保存了相似度和标签。cmp读取相似度进行比较

删除后会导致返回的结果错误，因此测试的map也不正确

**Comment by chj123456789** [comment]
但是不添加cmp，list.sort本身就可以根据相似度排序，我使用的是python3.6。

**Comment by chj123456789** [comment]
您好，我使用的是python3.6，按道理list.sort就可以返回汉明距离的排序结果，请问cmp读取的相似度指的是？

**Comment by burhanbarakli** [comment]
> this code is runing with python 2.7, the code might gives errors with pyrhon 3.
> 
> in our experiment, the best performance is 30,000-40,000 episodes on cifar dateset.

and
> 您好。cmp是比较函数，用于对检索结果的排序。list中保存了相似度和标签。cmp读取相似度进行比较
> 删除后会导致返回的结果错误，因此测试的map也不正确
 **i also translated this comment to english!**

>Hello. cmp is a comparison function, used to sort the search results.
>The similarity and label are saved in the list.cmp read similarity for comparison
>After deleting, the returned result will be wrong, so the tested map is not correct

_### so i understood the list.sort(cmp) is a necessary code line? am i right?_

**Comment by magicfisk** [comment]
> 

Yes!

Thanks for your attention. We will release a new code to fit python3 and provide the organized dataset with the pre-trained model in 2-3 days. I hope it can help you to better understand our paper.

**Comment by burhanbarakli** [comment]
> > 
> 
> Yes!
> 
> Thanks for your attention. We will release a new code to fit python3 and provide the organized dataset with the pre-trained model in 2-3 days. I hope it can help you to better understand our paper.

Thank you very much. I will wait new python 3 codes for your paper.
